crates/chia-bls/src/bls_cache.rs

```diff
@@ -17,16 +17,35 @@ use crate::{GTElement, PublicKey, Signature};
 /// However, validating a signature where we have no cached GT elements, the
 /// aggregate_verify() primitive is faster. When long-syncing, that's
 /// preferable.
+
+#[derive(Debug, Clone)]
+struct BlsCacheData {
+    // sha256(pubkey + message) -> GTElement
+    items: LinkedHashMap<[u8; 32], GTElement>,
+    capacity: NonZeroUsize,
+}
+
+impl BlsCacheData {
+    pub fn put(&mut self, hash: [u8; 32], pairing: GTElement) {
+        // If the cache is full, remove the oldest item.
+        if self.items.len() == self.capacity.get() {
+            if let Some((oldest_key, _)) = self.items.pop_front() {
+                self.items.remove(&oldest_key);
+            }
+        }
```

Comment:
```suggestion
        while self.items.len() > self.capacity.get() {
            let Some((oldest_key, _)) = self.items.pop_front() else { continue; };
            self.items.remove(&oldest_key);
        }
```
I'm not sure if this makes sense. It might be a bit more defensive